### PR TITLE
"Measures" formulas get options from CODAP (PT-186384867 + PT-186022282)

### PR DESCRIPTION
**Pull request by lublagg**
This PR updates the Measures formulas to create options based on:
-  The variables created inside the device (if we aren't in 'collector' mode)
-  If we are in 'collector' mode, all of the possible variables shared with the Sampler by the linked CODAP dataContext
- Any user-created attributes + their corresponding values in the Sampler-created CODAP table

### CODE REVIEW

File: TP-Sampler/src/app.js [hunk 1]
```diff
@@ -666,97 +666,133 @@ function setup() {
...
-    selectEl.append(outputOption);
-  }
+  // get list of attributes from CODAP
+  codapCom.getAttributesFromTable().then((attrs) => {
+    const attrNames = attrs.map((attr) => attr.name);
+    codapCom.getAllItems().then((items) => {
+      let attrMap = {};
+      attrNames.forEach((attrName) => {
+        if (attrName === deviceName && !isCollector) {
+          attrMap[attrName] = [...new Set(variables)];
```
Comment (dougmartin) [inline TP-Sampler/src/app.js:676]: is the `[...new Set(variables)]` there to filter out duplicate variable values?

File: TP-Sampler/src/app.js [hunk 2]
```diff
@@ -666,97 +666,133 @@ function setup() {
...
+            operatorOpt.textContent = option;
+            if (i === 0) {
+              operatorOpt.selected = true;
+            }
+            selectEl.appendChild(operatorOpt);
+          })
+        } else {
+          const equalsOpt = document.createElement("option");
+          equalsOpt.value = "=";
+          equalsOpt.textContent = "=";
```
Comment (dougmartin) [inline TP-Sampler/src/app.js:736]: Should `equalsOpt.selected = true` be set to match the other cases?

**Review by dougmartin — APPROVED** [review]
Looks good :+1:

File: TP-Sampler/src/app.js (hunk 1 above)
Comment (lublagg) [reply TP-Sampler/src/app.js:676]: Yes, exactly

File: TP-Sampler/src/app.js (hunk 2 above)
Comment (lublagg) [reply TP-Sampler/src/app.js:736]: I think it's OK as is, it seems to work as intended.

**Pull request merged**